### ISSUE TITLE
[Feature Request] Provide cron.php file

### ISSUE THREAD

**Issue by onny** [InvoicePlane/InvoicePlane-v2#395]
Hey,
Nextcloud provides a cron.php file in the web root directory which can be executed via php-cli on server side to trigger web cron.
This would be an easy alternative compared to manually access the full url as `http://your-domain.com/invoices/cron/recur/your-cron-key-here`.
In case I auto deploy a InvoicePlane instance, I could just enable a systemd-timer which executes the php script to update Invoiceplane tasks. Currently I have to manually put figure out the URL and cron key to use curl inside a systemd-timer job.

Best regards
Jonas

**Comment by nielsdrost7** [comment]
Hey @onny I overlooked your question, my apologies.
Come talk on Discord if you like: https://discord.gg/PPzD2hTrXt

Can't you select the cron key url from the database through your php-cli?

I'll try to resolve this for **v2** for sure. That's made with Laravel, so that's easier.

**Comment by nielsdrost7** [comment]
v2 uses **Laravel's artisan command scheduler** — there is no `cron.php` file or HTTP cron URL. Recurring invoice processing is (or should be) an artisan command that can be called directly or registered in `Kernel.php`/`routes/console.php`.

**What's already true in v2:**
- Laravel's scheduler: `php artisan schedule:run` is the single entry point — add this to system cron once: `* * * * * cd /path && php artisan schedule:run >> /dev/null 2>&1`.
- No secret cron key needed.
- Docker/systemd examples should be in the deployment docs.

**What still needs to be done:**
- Create artisan command: `php artisan invoiceplane:process-recurring` (if not already exists).
- Register it in the scheduler: `Schedule::command('invoiceplane:process-recurring')->daily()`.
- Document the command in the README / deployment guide.
- Add `--dry-run` flag for testing without actually creating invoices.

**Test Scenarios (PHPUnit):**
```php
// Artisan command processes due recurring invoices
/* Arrange */ RecurringInvoice::factory()->for($this->company)
                  ->create(['next_invoice_date' => today(), 'recurring_invoice_status' => 'active']);
/* Act */     Artisan::call('invoiceplane:process-recurring');
/* Assert */  $this->assertDatabaseHas('invoices', ['recurring_invoice_id' => RecurringInvoice::first()->id]);
```
```php
// --dry-run flag does not create invoices
/* Arrange */ RecurringInvoice::factory()->for($this->company)
                  ->create(['next_invoice_date' => today()]);
/* Act */     Artisan::call('invoiceplane:process-recurring', ['--dry-run' => true]);
/* Assert */  $this->assertDatabaseCount('invoices', 0);
```

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by nielsdrost7** [comment]
## Planning Note — v2 Implementation

**Story**
As a system administrator, I want to trigger the recurring invoice generator via an Artisan command so that I can schedule it with a system cron, systemd timer, or Laravel's task scheduler — without needing to know or maintain a secret URL and cron key.

**v2 Context**
In v2 (Laravel 11), the natural home for this is a dedicated Artisan command in `Modules/Invoices` registered with Laravel's console kernel. The `RecurringInvoice` model and its generation logic already live in `Modules/Invoices`. The command calls the existing `RecurringInvoiceService` (or equivalent service) to generate due invoices, making it composable. The command is also wired into `routes/console.php` (or `bootstrap/app.php` schedule) so it can run automatically via `php artisan schedule:run`.

**Acceptance Criteria**
- [ ] An Artisan command `invoices:process-recurring` exists in `Modules/Invoices`.
- [ ] Running `php artisan invoices:process-recurring` generates all recurring invoices that are due.
- [ ] The command outputs a summary line, e.g. `Generated 3 invoice(s) from recurring schedules.`
- [ ] The command accepts an optional `--dry-run` flag that reports how many invoices would be generated without creating any records.
- [ ] The command is registered in the Invoices module service provider (or `Console/Kernel.php`).
- [ ] The command is scheduled to run daily in `bootstrap/app.php` (or `routes/console.php`) via `->daily()`.
- [ ] No secret URL or cron key is required; the old HTTP-based trigger is not ported to v2.
- [ ] Feature is covered by PHPUnit tests.

**Implementation**
- Module: `Modules/Invoices/`
- New command: `Modules/Invoices/Console/Commands/ProcessRecurringInvoices.php`
  - `protected $signature = 'invoices:process-recurring {--dry-run : List due recurring invoices without creating them}'`
  - Calls `RecurringInvoiceService::processDue(dryRun: $this->option('dry-run'))`
  - Outputs count and exits 0
- `Modules/Invoices/Services/RecurringInvoiceService.php`: ensure a `processDue(bool $dryRun = false): int` method exists that queries `RecurringInvoice` records due for generation and either creates Invoice records or just counts them
- `Modules/Invoices/Providers/InvoicesServiceProvider.php`: register command via `$this->commands([ProcessRecurringInvoices::class])`
- `bootstrap/app.php` or `routes/console.php`: `Schedule::command('invoices:process-recurring')->daily()`

**Test Scenarios (PHPUnit)**
```php
// Extends AbstractCompanyPanelTestCase (needs DB + company)
public function test_process_recurring_command_generates_due_invoices(): void
{
    // Arrange — create a recurring invoice that is due today
    $recurring = RecurringInvoice::factory()->for($this->company)->create([
        'next_due_date' => now()->toDateString(),
        'is_active'     => true,
    ]);

    // Act
    $this->artisan('invoices:process-recurring')
         ->assertExitCode(0)
         ->expectsOutputToContain('Generated 1 invoice(s)');

    // Assert
    $this->assertDatabaseHas('invoices', ['recurring_invoice_id' => $recurring->id]);
}

public function test_process_recurring_command_dry_run_does_not_create_invoices(): void
{
    // Arrange
    RecurringInvoice::factory()->for($this->company)->create([
        'next_due_date' => now()->toDateString(),
        'is_active'     => true,
    ]);

    // Act
    $this->artisan('invoices:process-recurring', ['--dry-run' => true])
         ->assertExitCode(0)
         ->expectsOutputToContain('1 invoice(s) would be generated');

    // Assert — nothing was actually created
    $this->assertDatabaseCount('invoices', 0);
}

public function test_process_recurring_command_skips_invoices_not_yet_due(): void
{
    // Arrange
    RecurringInvoice::factory()->for($this->company)->create([
        'next_due_date' => now()->addDays(5)->toDateString(),
        'is_active'     => true,
    ]);

    // Act
    $this->artisan('invoices:process-recurring')
         ->assertExitCode(0)
         ->expectsOutputToContain('Generated 0 invoice(s)');

    // Assert
    $this->assertDatabaseCount('invoices', 0);
}
```

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by nielsdrost7** [comment]
---

## Test Suggestions

_Added to support future implementation of this request. The issue body is unchanged._

`Modules\Invoices\Services\RecurringInvoiceService` is currently an empty stub class (`class RecurringInvoiceService {}` — no methods at all), and there is no console command anywhere under `Modules/Invoices` (`find Modules/Invoices -ipath "*Console*"` returns nothing) and no scheduler entry in `routes/console.php` (which only defines the default `inspire` command). So there is neither a web cron URL nor an artisan command today — this request and the underlying generation logic (tracked separately, see the recurring-invoice profile epic) need to land together: the artisan command this issue asks for is only useful once `RecurringInvoiceService` actually generates invoices.

- **PHPUnit:** `it_registers_a_recurring_invoices_artisan_command()`
```php
#[Test]
public function it_registers_a_recurring_invoices_artisan_command(): void
{
    /* Act */
    $exitCode = Artisan::call('invoices:generate-recurring');

    /* Assert */
    $this->assertSame(0, $exitCode);
}
```

- **PHPUnit:** `it_generates_due_recurring_invoices_when_the_command_runs()`
```php
#[Test]
public function it_generates_due_recurring_invoices_when_the_command_runs(): void
{
    /* Arrange */
    $recurring = RecurringInvoice::factory()->for($this->company)->create([
        'start_at' => now()->subDay()->toDateString(),
    ]);

    /* Act */
    Artisan::call('invoices:generate-recurring');

    /* Assert */
    $this->assertDatabaseHas('invoices', ['recurring_invoice_id' => $recurring->id]);
}
```

- **PHPUnit:** `it_does_not_require_a_cron_key_or_url_when_run_via_artisan()`
```php
#[Test]
public function it_does_not_require_a_cron_key_or_url_when_run_via_artisan(): void
{
    /* Act */
    $exitCode = Artisan::call('invoices:generate-recurring');

    /* Assert */
    $this->assertSame(0, $exitCode);
}
```

**Edge cases worth covering**: running the command when no recurring profiles are due (should exit cleanly, no error), and running it twice in the same period (should not double-generate — the same idempotency concern already flagged against the underlying `RecurringInvoiceService::generateDueInvoices()` work).

**Comment by nielsdrost7** [comment]
> Original request (preserved): Nextcloud provides a cron.php file which can be executed via php-cli to trigger web cron — an easy alternative to manually accessing the full URL `http://your-domain.com/invoices/cron/recur/your-cron-key-here`. For auto-deployed instances a systemd-timer should be able to run a CLI entry point instead of curl-ing a URL with a secret key.

In v2 this becomes the native trigger for the recurring generation engine: v1's public endpoint `invoices/cron/recur/{cron_key}` (`application/modules/invoices/controllers/Cron.php:22-31`, guarded by `hash_equals` against the `cron_key` setting) is replaced entirely by an artisan command plus Laravel's scheduler. No HTTP endpoint, no shared secret.

### Specific
1. Add `Modules/Invoices/Console/GenerateRecurringInvoicesCommand.php`, signature `invoices:generate-recurring {--company= : restrict to one company id} {--dry-run}`, which calls `GenerateRecurringInvoicesService::generateDue()` (engine issue under #52) across all companies and reports `<n> recurring invoices processed` (v1 log line, `Cron.php:183`).
2. Register it in `InvoicesServiceProvider::registerCommands()` and schedule it daily in `registerCommandSchedules()` (both currently commented out, `Modules/Invoices/Providers/InvoicesServiceProvider.php:72-83`).
3. Remove the vestigial `cron_key` setting field and its hard-coded default `'R83fys4wWoNuUXtv'` from `Modules/Core/Filament/Admin/Pages/Settings.php:46,277-285` — nothing consumes it and a keyed public URL is not part of the v2 design.

### Measurable
- `php artisan invoices:generate-recurring` exits 0 and prints a per-company summary.
- `php artisan schedule:list` shows the command scheduled daily.
- `--dry-run` lists due schedules without writing.
- No route or controller exposes generation over HTTP.

### Achievable
Thin wrapper: all logic lives in the service (engine issue); the command only iterates companies and prints. Scheduler wiring is the standard `$schedule->command('invoices:generate-recurring')->daily()` inside the module's `registerCommandSchedules()`.

### Relevant
Deployment story for every self-hosted instance: one `schedule:run` cron entry replaces v1's per-install URL + secret. Directly requested by users migrating from v1 automation.

### Time-Bound
Half a sprint once the generation engine service lands; part of Epic #52.

#### Arrange
- Two companies, each with one due recurring schedule and one not-due schedule (engine test fixtures).

#### Act
- Run `php artisan invoices:generate-recurring`; then `php artisan invoices:generate-recurring --company=<first>` the next cycle.

#### Assert
- [ ] First run creates exactly one invoice per company; output names both companies with counts.
- [ ] `--company=` run touches only that company's schedules.
- [ ] `--dry-run` writes nothing and lists the due schedules.
- [ ] Command is listed by `php artisan schedule:list` (daily).
- [ ] `cron_key` no longer appears in the admin Settings page or seeded defaults.
